### PR TITLE
Merge from Drive (Turret Cleanup) Development Branch

### PR DESCRIPTION
**Pull request by qu-gg**
Proposing standardized way of adding objects to the ship - creating an object and passing that object directly into the ship.

Changes:
Added functionality for adding/removing turrets to a ship, added relevant tests in test_spacecraft.py
Cleaned documentation and style in turrets.py

### CODE REVIEW

File: shipyard/models/turrets.py [hunk 1]
```diff
@@ -15,17 +15,20 @@ class Turret:
...
+        self.weapons         = list()                              # list of the weapons on this turret
+        self.cost            = self.model.get("cost")              # cost of the turret and addons
 
     def add_addon(self, part):
-        # Grabbing addon info from data
+        """
+        Handles adding an add-on to the turret
+        :param part: Name of the add-on to add
+        """
         addon = self.data.get("addons").get(part)
```
Comment (Milkshak3s) [inline shipyard/models/turrets.py]: Shouldn't we use .append() here, as addons is a list?

File: shipyard/models/spacecraft.py [hunk 2]
```diff
@@ -126,3 +127,24 @@ def add_pplant(self, plant_type):
         self.cargo = self.cargo - new_pplant.tonnage
         self.fuel_two_weeks = new_pplant.fuel_two_weeks
 
+    def add_turret(self, turret):
```
Comment (Milkshak3s) [inline shipyard/models/spacecraft.py]: Should we check here if turret exceeds max hardpoints? Or just let things overrun and display in the red on GUI?

(legit question i haven't thought about it yet)

Comment (qu-gg) [reply shipyard/models/spacecraft.py]: To be honest, I hadn't actually considered hardpoints yet. Letting things overrun and letting an overarching handler might be more appropriate.
I'd reckon sticking with the latter option.

File: shipyard/models/turrets.py (hunk 1 above)
Comment (qu-gg) [reply shipyard/models/turrets.py]: This is grabbing the add-on data from the json file before appending it later down the function in line 35.

Comment (Milkshak3s) [reply shipyard/models/turrets.py]: Oh it's not self.addon :P I'm dumb

**Comment by Milkshak3s** [comment]
lgtm

**Pull request merged**